File: send_whatsapp.py
import sys
import mysql.connector
import pywhatkit as kit
from datetime import datetime, timedelta

# Database connection details
db_host = 'localhost'
db_user = 'root'  # Your MySQL username
db_password = 'Bhavani@2005'  # Your MySQL password
db_name = 'donnor_registration1'  # Replace with your database name

# Connect to the database
def connect_to_db():
    try:
        connection = mysql.connector.connect(
            host=db_host,
            user=db_user,
            password=db_password,
            database=db_name
        )
        return connection
    except mysql.connector.Error as err:
        print(f"Error: {err}")
        return None

# Fetch phone numbers for the selected blood group
def fetch_donors_by_blood_group(blood_group):
    connection = connect_to_db()
    if connection:
        cursor = connection.cursor(dictionary=True)
        query = "SELECT name, whatsapp FROM donors WHERE blood_group = %s"
        cursor.execute(query, (blood_group,))
        donors = cursor.fetchall()
        connection.close()

        return donors
    else:
        return []

# Function to send WhatsApp message
def send_whatsapp_message(phone, message, send_time):
    try:
        # Send WhatsApp message using pywhatkit
        kit.sendwhatmsg(phone, message, send_time.hour, send_time.minute, wait_time=20)
    except Exception as e:
        print(f"Failed to send message to {phone}: {e}")

# Function to schedule messages
def send_alerts_for_blood_group(blood_group):
    # Fetch the donors for the selected blood group
    donors = fetch_donors_by_blood_group(blood_group)

    if donors:
        print(f"Sending messages to donors with blood group {blood_group}...\n")

        # Get current time and calculate the send time
        now = datetime.now()
        send_time = now + timedelta(minutes=1, seconds=15)  # Add 1 minute and 15 seconds to the current time

        for donor in donors:
            name = donor['name']
            phone = donor['whatsapp']  # Ensure the phone number includes the country code (e.g., +91)

            # Message to send
            message = f"Dear {name},\nWe urgently need your blood group {blood_group}. Please contact us immediately if available. Thank you for your support!"

            # Send the message to each donor
            send_whatsapp_message(phone, message, send_time)
    else:
        print(f"No donors found for blood group {blood_group}.")

# Main execution
if __name__ == "__main__":
    # Get the blood group from the command line argument passed from PHP
    if len(sys.argv) > 1:
        blood_group_to_search = sys.argv[1]
        send_alerts_for_blood_group(blood_group_to_search)
    else:
        print("No blood group provided!")

File: send_whatsapp.php
<?php
require_once __DIR__ . '/vendor/autoload.php'; // Correct the path as needed
include 'config.php';

use Twilio\Rest\Client;

// Twilio credentials
$account_sid = 'ACe468e0826d73702b057f0b5341a48e68';
$auth_token =  '0ab6c11c95e336adababc7ee38e5a1e7';
$twilio_number = 'whatsapp:+14155238886';

// Fetch the selected blood group
$blood_group = $_POST['blood_group'];

// Get donors with the selected blood group
$query = "SELECT name, phone FROM donors WHERE blood_group = ?";
$stmt = $conn->prepare($query);
$stmt->bind_param("s", $blood_group);
$stmt->execute();
$result = $stmt->get_result();

if ($result->num_rows > 0) {
    $client = new Client($account_sid, $auth_token);

    while ($donor = $result->fetch_assoc()) {
        $name = $donor['name'];
        $phone = 'whatsapp:+' . $donor['phone']; // Ensure phone numbers include country code

        $message = "Dear $name,\nWe urgently need your blood group $blood_group. If available, please contact us immediately.\nThank you for your support!";

        try {
            $client->messages->create($phone, [
                'from' => $twilio_number,
                'body' => $message,
            ]);
        } catch (Exception $e) {
            echo "Error sending to $phone: " . $e->getMessage();
        }
    }

    echo "Alerts sent successfully!";
} else {
    echo "No donors found for blood group $blood_group.";
}
?>